sdk/src/wallet/operations/helpers/time.rs

```diff
@@ -24,13 +26,16 @@ pub(crate) fn can_output_be_unlocked_now(
         .required_address(commitment_slot_index.into(), committable_age_range)?;
 
     // In case of `None` the output can currently not be unlocked because of expiration unlock condition
-    Ok(required_address.map_or_else(|| false, |required_address| wallet_address == &required_address))
+    Ok(required_address.map_or_else(
+        || false,
+        |required_address| controlled_addresses.contains(&required_address),
+    ))
 }
 
 // Check if an output can be unlocked by the wallet address at the current time and at any
 // point in the future
 pub(crate) fn can_output_be_unlocked_forever_from_now_on(
```

Comment:
Can we just remove `forever` from this name as it is not necessary

Reply:
Removed